database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Models/Post.swift

```diff
@@ -14,33 +14,35 @@
 //  limitations under the License.
 //
 
-import UIKit
+import Foundation
 import Firebase
 
-struct Post {
+struct Post: Identifiable {
+  var id: String
   var uid: String
   var author: String
   var title: String
   var body: String
   var starCount: Int
-  var stars: [String: Bool]?
 
-  init(uid: String, author: String, title: String, body: String) {
+  init(id: String, uid: String, author: String, title: String, body: String) {
+    self.id = id
     self.uid = uid
     self.author = author
     self.title = title
     self.body = body
     starCount = 0
   }
 
-  init?(snapshot: DataSnapshot) {
-    guard let dict = snapshot.value as? [String: Any] else { return nil }
+  init?(dict: [String: Any]) {
+    guard let id = dict["id"] as? String else { return nil }
```

Comment:
Instead of doing this, you could rewrite the initializer to take two parameters:

```swift
init?(id: String, dict: [String: Any]) { /* ... */ }
```

Then, later on, the `id` doesn't need to be a part of the dictionary, which will save you some storage space in the database.

Reply:
I feel that the "id" information would be helpful to retrieve a specific post and make modifications based on it. Such as incrementing stars and comments. Or is there another way to do so without the id information?

Reply:
Actually I see that the id will still be included as part of the post object in the last step. I see, thanks for pointing it out!